### PR TITLE
 using opentelemetry.io* namespace for extracing resource based on annotations  from k8s Api #2330 

### PR DESCRIPTION
**Pull request by Horiodino**
change_type: sdk.go

- Modified the logic for iterating over pod annotations, utilizing pod.GetAnnotations() to identify and extract resources using the specified namespace prefix (constants.ReservedNamespace).
 - Implemented the use of the OpenTelemetry.io namespace for extracting resource annotations from Kubernetes pods.

- Enhanced handling to streamline the extraction process and avoid overwriting existing resource mappings.

issues: [https://github.com/open-telemetry/opentelemetry-operator/issues/2181]

https://github.com/open-telemetry/opentelemetry-operator/pull/2330

TestCase :white_check_mark: 
```
?       github.com/open-telemetry/opentelemetry-operator/pkg/constants  [no test files]
ok      github.com/open-telemetry/opentelemetry-operator/pkg/collector/upgrade  
ok      github.com/open-telemetry/opentelemetry-operator/pkg/featuregate        
ok      github.com/open-telemetry/opentelemetry-operator/pkg/instrumentation    
ok      github.com/open-telemetry/opentelemetry-operator/pkg/instrumentation/upgrade    
ok      github.com/open-telemetry/opentelemetry-operator/pkg/sidecar    ```

### CODE REVIEW

**Comment by swiatekm-sumo** [comment]
The title of your PRs indicates it's adding support for attribute labels as well resource ones, but the code only seems to implement resources. Are you planning on a follow-up?

**Comment by Horiodino** [comment]
oh sorry for that , i have just added  anotation for now  because annotations are for attaching non-identifying metadata. 
well @jaronoff97  should i add labels too ?

**Comment by jaronoff97** [comment]
I think we should only do from annotations and not labels. @Horiodino can you add a changelog and update the title to specify only annotations?

**Comment by Horiodino** [comment]
done

File: pkg/constants/env.go [hunk 1]
```diff
@@ -31,11 +31,12 @@ const (
...
-	EnvPodUID   = "OTEL_RESOURCE_ATTRIBUTES_POD_UID"
-	EnvPodIP    = "OTEL_POD_IP"
-	EnvNodeName = "OTEL_RESOURCE_ATTRIBUTES_NODE_NAME"
-	EnvNodeIP   = "OTEL_NODE_IP"
+	EnvPodName        = "OTEL_RESOURCE_ATTRIBUTES_POD_NAME"
+	EnvPodUID         = "OTEL_RESOURCE_ATTRIBUTES_POD_UID"
+	EnvPodIP          = "OTEL_POD_IP"
+	EnvNodeName       = "OTEL_RESOURCE_ATTRIBUTES_NODE_NAME"
+	EnvNodeIP         = "OTEL_NODE_IP"
+	ReservedNamespace = "resource.opentelemetry.io/"
```
Comment (swiatekm-sumo) [inline pkg/constants/env.go:39]: How about we rename this to `OtelAnnotationNamespace`? Just `ReservedNamespace` runs the risk of it being confused for a normal K8s Namespace.

**Review by swiatekm-sumo — APPROVED** [review]
LGTM, small nitpick about naming. Please add a changelog entry as well, you can run `make chlog-new` to have the tooling generate a template for you.

File: pkg/constants/env.go (hunk 1 above)
Comment (Horiodino) [reply pkg/constants/env.go]: done

**Comment by jaronoff97** [comment]
@Horiodino last thing to do here is add a changelog via `make chlog-new`. That will generate a new file in the .chloggen/ You can then edit that file with the necessary details at which point we can merge this :D

**Comment by Horiodino** [comment]
@jaronoff97  hearding about changelog  first time ! i  added it now

File: .chloggen/extract-resources-ns.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+change_type: enhancement
+
+component: instrumentation
+
+note: "Modified the logic for iterating over pod annotations, utilizing pod.GetAnnotations() to identify and extract resources using the specified namespace prefix (constants.OtelAnnotationNamespace)."
+
+issues:
```
Comment (jaronoff97) [inline .chloggen/extract-resources-ns.yaml:7]: This can just be the issue number

File: .chloggen/extract-resources-ns.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+change_type: enhancement
+
+component: instrumentation
+
+note: "Modified the logic for iterating over pod annotations, utilizing pod.GetAnnotations() to identify and extract resources using the specified namespace prefix (constants.OtelAnnotationNamespace)."
```
Comment (jaronoff97) [inline .chloggen/extract-resources-ns.yaml:5]: I would say "introduced ability to set Otel resource attributes based on annotations for instrumentation"

File: .chloggen/extract-resources-ns.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+issues:
+  - https://github.com/open-telemetry/opentelemetry-operator/issues/2181
+
+
+subtext: |
```
Comment (jaronoff97) [inline .chloggen/extract-resources-ns.yaml:11]: I would just mention the syntax for setting one of the attributes here

**Review by jaronoff97 — APPROVED** [review]

**Review by pavolloffay — COMMENTED** [review]
Can we please document this in the root readme?

**Comment by jaronoff97** [comment]
@Horiodino when you get a chance, if you could add some documentation into the README.md then we should be able to merge this! :bow:

File: README.md [hunk 5]
```diff
@@ -712,6 +712,25 @@ spec:
 EOF
 ```
 
+### using opentelemetry.io* namespace for extracing resource
```
Comment (swiatekm-sumo) [inline README.md:715]:
```suggestion
### Setting instrumentation resource attributes via namespace annotations
```

File: README.md [hunk 6]
```diff
@@ -712,6 +712,25 @@ spec:
...
+
+This example shows a pod configuration with OpenTelemetry annotations using the `resource.opentelemetry.io/` prefix. These annotations can be used to extract resource information for OpenTelemetry instrumentation.
```
Comment (swiatekm-sumo) [inline README.md:717]:
```suggestion
This example shows a pod configuration with OpenTelemetry annotations using the `resource.opentelemetry.io/` prefix. These annotations can be used to add resource attributes to data produced by OpenTelemetry instrumentation.
```

**Review by swiatekm-sumo — COMMENTED** [review]
Suggested some phrasing changes to the documentation.

**Review by jaronoff97 — APPROVED** [review]

**Comment by jaronoff97** [comment]
@Horiodino thank you for your contribution :D

**Pull request merged**